backend/src/main/java/instantPolls/controller/RestControllerImpl.java

```diff
@@ -70,18 +67,29 @@
 	
 	@ResponseBody
 	@GetMapping(value = "/room/{id}")
-	public Room getRoom(@PathVariable(value = "id") String id) {
+	public ResponseEntity<Room> getRoom(@PathVariable(value = "id") String id) {
 		Room room = roomStorage.findRoomById(id);
-		return room;
+		if(room != null)
+			return ResponseEntity.ok(room);
+		else

```

Comment:
return ResponseEntity.ok(room) is all you need here. Btw consider using Optionals in such situation (it would be wiser to have roomStorage return Optional.empty if no room was found.
See https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html